include/triton/backend/backend_common.h

```diff
@@ -671,4 +672,23 @@ TRITONSERVER_Error* BufferAsTypedString(
 /// \return a formatted string for logging the request ID.
 std::string GetRequestId(TRITONBACKEND_Request* request);
 
+/// Validate the contiguous string buffer with correct format
+/// <int32_len><bytes>...<int32_len><bytes>.
+/// @param buffer The pointer to the contiguous string buffer.
+/// @param buffer_byte_size The size of the buffer in bytes.
+/// @param expected_element_cnt The number of expected string elements.
+/// @param input_name The name of the input buffer.
+/// @param element_idx Returns the number of validated strings.
+/// @param set_string_tensor_cb Callback function which sets string input tensor
+/// depending on the backend platform.
+/// @param onnx_backend Whether the backend platform is ONNX runtime.
+/// @return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_Error* ValidateStringBuffer(
+    const char* buffer, size_t buffer_byte_size,
+    const size_t expected_element_cnt, const char* input_name,
+    size_t* element_idx,
+    const std::function<void(size_t, const char*, const uint32_t)>&
```

Comment:
I don't like how this callback function is passed into `ValidateStringBuffer` in order to add strings from buffer into the string tensor. Open to suggestions.

Reply:
Based on discussion with Tanmay, I have separated the logic of setting string tensor from validating string buffer.